### PR TITLE
Fix #28973: Disable drag for single fields

### LINKED ISSUE #28973: Chart Settings allows you to drag fields when there is only 1 item in the list
Dragging when you have only 1 metric or 1 dimension should be disabled. Today, it allows you to drag the field around, though you can't place it anywhere

**To Reproduce**
1. Go to New -> Question -> Orders
2. Summarize: Count by Created At: Year
3. Visualize and open chart settings. Notice that Count and Created At can be dragged around.

**Expected behavior**
Those fields should only be draggable if there is more than 1 element in the list

chrome_InV7t1KNAH

### PR DESCRIPTION
**Pull request by JesseSDevaney**
**Problem:** When there is only one field/sub-field, they are still draggable

Fields
draggable-fields-problem

Sub-Fields
draggable-fields-sub-fields-problem

**Context:** 
- Closes #28973

**Desired Behavior:** 
- [ ] A field should be draggable when there is > 1 fields in the group
  - [ ] A sub-field should be draggable when there is > 1 sub-fields in the group
- [ ] A field should not be draggable when there is <= 1 fields in the group
  - [ ] A sub-field should not be draggable when there is <= 1 sub-fields in the group

**Solution Implementation:**
- Check the number of fields and set a field to draggable if there is more than one in a group and not draggable if there is one or less

Fields
draggable-fields-solution

Sub-Fields
draggable-fields-sub-fields-solution

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30408)

### CODE REVIEW

**Review by WiNloSt — APPROVED** [review]
Looks great. I think maybe there is another thing worth following-up though. The sub-dimension is still draggable even when there's only 1 sub-dimension.



> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30408#-:-NU1oSvy8Jic4pgYWZYn:bx4e56q)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @JesseSDevaney)

___
*[`frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx` line 33 at r1](https://reviewable.io/reviews/metabase/metabase/30408#-NU1n3I63cyRIzCcBsQy:-NU1n3I63cyRIzCcBsQz:b-6z6qw) ([raw file](https://github.com/metabase/metabase/blob/d1a67bbfecdc1a511c22beb6b53a9a687436b2a5/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx#L33)):*
> ```JavaScript
>   };
> 
>   const isDragDisabled = Array.isArray(fields) && fields.length <= 1;
> ```

nit: could this be more concise?
```ts
const isDragDisabled = fields?.length <= 1;
```

In my opinion, `Array.isArray` would be really useful when we're dealing with variables that could be something else other than an array. e.g.
https://github.com/metabase/metabase/blob/d1a67bbfecdc1a511c22beb6b53a9a687436b2a5/frontend/src/metabase/actions/components/ActionViz/utils.ts#L20

But in this case, these 2 settings that we are dragging are `graph.dimensions` and `graph.metrics` which as far as I can tell, are always an array @npfitz maybe you can confirm that?

**Comment by npfitz** [comment]
*[`frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx` line 33 at r1](https://reviewable.io/reviews/metabase/metabase/30408#-NU1n3I63cyRIzCcBsQy:-NU1q44S80JtRrutBN92:b-5azttw) ([raw file](https://github.com/metabase/metabase/blob/d1a67bbfecdc1a511c22beb6b53a9a687436b2a5/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx#L33)):*
<details><summary><i>Previously, WiNloSt (Mahatthana (Kelvin) Nomsawadi) wrote…</i></summary><blockquote>

nit: could this be more concise?
```ts
const isDragDisabled = fields?.length <= 1;
```

In my opinion, `Array.isArray` would be really useful when we're dealing with variables that could be something else other than an array. e.g.
https://github.com/metabase/metabase/blob/d1a67bbfecdc1a511c22beb6b53a9a687436b2a5/frontend/src/metabase/actions/components/ActionViz/utils.ts#L20

But in this case, these 2 settings that we are dragging are `graph.dimensions` and `graph.metrics` which as far as I can tell, are always an array @npfitz maybe you can confirm that?
</blockquote></details>

Yes, dimensions and metrics should always be an array

File: frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx [hunk 1]
```diff
@@ -30,6 +30,8 @@ const ChartSettingFieldsPicker = ({
     );
   };
 
+  const isDragDisabled = Array.isArray(fields) && fields.length <= 1;
```
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx:33]: I believe `fields.length === 1` will work as well, if length is 0, `map` will render nothing

**Comment by JesseSDevaney** [comment]
I added in functionality to prevent sub-dimension dragging when there is only one item in the sub-dimension.

**Review by WiNloSt — APPROVED** [review]
Works great. And I left some small notes that could make the code more coherent.

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30408#-:-NU7-FEkCoSEqTIW2isL:bmp0ypw)* status: all files reviewed, 4 unresolved discussions (waiting on @JesseSDevaney, @npfitz, and @uladzimirdev)

___
*[`frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx` line 33 at r1](https://reviewable.io/reviews/metabase/metabase/30408#-NU1n3I63cyRIzCcBsQy:-NU6vKh_856Um3ZfQWvd:bwpyks9) ([raw file](https://github.com/metabase/metabase/blob/d1a67bbfecdc1a511c22beb6b53a9a687436b2a5/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx#L33)):*
<details><summary><i>Previously, npfitz (Nick Fitzpatrick) wrote…</i></summary><blockquote>

Yes, dimensions and metrics should always be an array
</blockquote></details>

Then maybe this line should be updated to reflect our conversation too.
https://github.com/metabase/metabase/blob/f20fab5bfd911c57b10e8544770df1ad6bc3b42f/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx#L35

Changing this to like `fields?.length >= 1` would make it more consistent with

```ts
 const isDragDisabled = fields?.length === 1;
```
___
*[`frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedItems.tsx` line 83 at r2](https://reviewable.io/reviews/metabase/metabase/30408#-NU6w5rY0snhPqA_Cq95:-NU6w5rY0snhPqA_Cq96:b296lq0) ([raw file](https://github.com/metabase/metabase/blob/4bf181b91f7b004195fd6ecc109a47755c6f671f/frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedItems.tsx#L83)):*
> ```XML
>   onColorChange,
> }: SortableColumnListProps<T>) {
>   const isDragDisabled = items?.length === 1;
> ```

nit: From its type and also the usage down below. It suggests that `items` will never be `undefined` or `null`. So I think we should remove optional chaining here since that kind of suggest `items` would be `undefined` or `null`
___
*[`frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedItems.tsx` line 99 at r2](https://reviewable.io/reviews/metabase/metabase/30408#-NU6zaod4TKhKFe0N7fl:-NU6zaoe5IRaM_O8vEVS:bexxeyc) ([raw file](https://github.com/metabase/metabase/blob/4bf181b91f7b004195fd6ecc109a47755c6f671f/frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedItems.tsx#L99)):*
> ```XML
>           onColorChange={onColorChange}
>           disabled={isDragDisabled}
>           isDragDisabled={isDragDisabled}
> ```

This could confuse some people why we need 2 flags. I have no good answer for this, but seems like `disabled` is for [`react-sortable-hoc`](https://github.com/clauderic/react-sortable-hoc#sortableelement-hoc) and it doesn't pass this prop down to the component being wrapped in `SortableElement`.

That is why we need to pass another flag, `isDragDisabled` in this case, so we could use it to determine when to show the drag handle.

Ideally, I'd expect passing only 1 boolean flag especially when they achieve the same thing.



_Code quote:_
```XML
          disabled={isDragDisabled}
          isDragDisabled={isDragDisabled}
```

**Review by npfitz — COMMENTED** [review]
I agree with @WiNloSt on the `disabled` and `isDraggingDisbaled`, but otherwise, it looks good to me.

> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30408#-:-NU7004E6UQJ2Hkaz0jH:b-vwul1o)* status: all files reviewed, 4 unresolved discussions (waiting on @JesseSDevaney and @uladzimirdev)

**Review by npfitz — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30408)* status: all files reviewed, 4 unresolved discussions (waiting on @JesseSDevaney and @uladzimirdev)

**Pull request merged**